ContactDetailsApi/V2/UseCase/FetchAllContactDetailsByUprnUseCase.cs

```diff
@@ -100,16 +103,16 @@ public List<ContactByUprn> ConsolidateData(IEnumerable<TenureInformation> tenure
         }
 
         [LogCall]
-        public async Task<ContactsByUprnList> ExecuteAsync()
+        public async Task<PagedResult<ContactByUprn>> ExecuteAsync(ServicesoftFetchContactDetailsRequest request)
         {
-            var tenures = await GetTenures().ConfigureAwait(false);
-            var personIds = FilterPersonIds(tenures.ToList());
+            var tenures = await GetTenures(request.PaginationToken, request.PageSize).ConfigureAwait(false);
+            var personIds = FilterPersonIds(tenures.Results);
 
             var persons = await GetPersons(personIds);
             var contactDetails = await GetContactDetails(personIds);
```

Comment:
Ik this this is outside the scope of this PR. But could we use TaskList.whenAll([]) for these?

Reply:
We were considering that, and yes, that could help performance a lot. Might raise a separate PR for that